config/config_test.go

```diff
@@ -457,6 +469,50 @@ func TestTreeRootFile(t *testing.T) {
 	checkValue(tempDir, ".git/config")
 }
 
+func TestTreeRootCmd(t *testing.T) {
+	as := require.New(t)
+
+	cfg := &config.Config{}
+	v, flags := newViper(t)
+
+	checkValue := func(treeRoot string) {
+		readValue(t, v, cfg, func(cfg *config.Config) {
+			as.Equal(treeRoot, cfg.TreeRoot)
+		})
+	}
+
+	tempDir := t.TempDir()
+	as.NoError(os.MkdirAll(filepath.Join(tempDir, "foo"), 0o755))
+	as.NoError(os.MkdirAll(filepath.Join(tempDir, "bar"), 0o755))
+
+	// default with no flag, env or config
+	// should match the absolute path of the directory in which the config file is located
+	checkValue(filepath.Dir(v.ConfigFileUsed()))
+
+	// set config value
+	cfg.TreeRootCmd = "echo " + tempDir
+	checkValue(tempDir)
+
+	// env override
+	// should match the directory above
+	t.Setenv("TREEFMT_TREE_ROOT_CMD", fmt.Sprintf("echo \"%s/foo\"", tempDir))
+	checkValue(filepath.Join(tempDir, "foo"))
+
+	// flag override
+	// should match the root of the temp directory structure
+	as.NoError(flags.Set("tree-root-cmd", fmt.Sprintf("echo '%s/bar'", tempDir)))
+	checkValue(filepath.Join(tempDir, "bar"))
+
+	// empty tree-root-cmd value
+	// should be the same as the current working directory
+	// we switch to a new temp directory to test this
+	tempDir2 := t.TempDir()
+	t.Chdir(tempDir2)
+
+	as.NoError(flags.Set("tree-root-cmd", "echo ''"))
+	checkValue(tempDir2)
```

Comment:
I personally find this to be confusing behavior. If somebody writes a script for a `tree-root-cmd` and accidentally forgets to print something, wouldn't it be better for them to get an error, rather than having treefmt fallback to some undocumented behavior?

Reply:
Agreed, this should either be clearly documented or throw a user-facing error. An error is my preference.

Reply:
I have changed it to throw an error if the `tree-root-cmd` returns an empty output.